src/conf_helpers.c

```diff
@@ -0,0 +1,92 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <unistd.h>  
+#include <signal.h>
+
+#include "mosquitto_broker_internal.h"
+
+#include "conf_helpers.h"
+
+
+int update_conf_param(struct mosquitto__config *config, const char *param_key, const char *param_value) {
+    if(strcmp(param_key, "allow_anonymous") == 0) {
+        if(strcmp(param_value, "true") == 0) {
+            config->security_options.allow_anonymous = -1;
```

Comment:
-1?

Reply:
In initialization functions of conf.c, "-1" is used. At first I thought is is for "true", that's why I follow it. But now, when I double check it, I believe it is in the meaning of "not set". Changed it with "1".